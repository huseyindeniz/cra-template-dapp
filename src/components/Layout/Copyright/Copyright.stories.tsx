// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Copyright } from ".";
import {
  copyrightLabel,
  copyrightUrl,
  copyrightLogoUrl,
} from "../../../config";
export default {
  title: "Layout/Copyright",
  component: Copyright,
  args: {
    copyrightLabel: copyrightLabel,
    copyrightLogoUrl: copyrightLogoUrl,
    copyrightUrl: copyrightUrl,
  },
} as ComponentMeta<typeof Copyright>;

export const Default: ComponentStory<typeof Copyright> = (args) => (
  <Copyright {...args} />
);
