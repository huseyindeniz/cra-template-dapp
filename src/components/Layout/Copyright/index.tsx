import React from "react";
import { Box, Link, Avatar, Button } from "@chakra-ui/react";
import { FaExternalLinkAlt } from "@react-icons/all-files/fa/FaExternalLinkAlt";

export const Copyright: React.FC = React.memo(() => {
  const copyrightLabel: string = "dApp CRA Template";
  const copyrightUrl: string =
    "https://github.com/huseyindeniz/cra-template-dapp";
  const copyrightLogoUrl: string =
    "https://huseyindeniz.net/assets/images/logo.webp";

  return (
    <Box color={"grey"}>
      <Button
        as={Link}
        href={copyrightUrl}
        isExternal
        variant={"ghost"}
        size={"xs"}
        rightIcon={<FaExternalLinkAlt />}
        leftIcon={
          <Avatar size={"xs"} src={copyrightLogoUrl} name={copyrightLabel} />
        }
      >
        {copyrightLabel}
      </Button>
    </Box>
  );
});
