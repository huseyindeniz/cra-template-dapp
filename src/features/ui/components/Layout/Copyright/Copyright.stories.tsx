// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Copyright } from ".";
export default {
  title: "Layout/Copyright",
  component: Copyright,
} as ComponentMeta<typeof Copyright>;

export const Default: ComponentStory<typeof Copyright> = (args) => (
  <Copyright {...args} />
);
