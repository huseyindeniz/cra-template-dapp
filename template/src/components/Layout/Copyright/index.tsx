import React from "react";
import { Box, Link, Avatar, Button } from "@chakra-ui/react";
import { FaExternalLinkAlt } from "@react-icons/all-files/fa/FaExternalLinkAlt";

export const Copyright: React.FC = () => {
  const copyrightLabel: string = "dApp CRA Template";
  const copyrightUrl: string =
    "https://github.com/huseyindeniz/cra-template-dapp";
  const copyrightLogoUrl: string =
    "https://huseyindeniz.net/static/media/logo.cf1063933b90ff363f2d.jpg";

  return (
    <Box color={"grey"}>
      <Button
        as={Link}
        href={copyrightUrl}
        isExternal
        variant={"ghost"}
        size={"xs"}
        rightIcon={<FaExternalLinkAlt />}
        leftIcon={
          <Avatar size={"xs"} src={copyrightLogoUrl} name={copyrightLabel} />
        }
      >
        {copyrightLabel}
      </Button>
    </Box>
  );
};
