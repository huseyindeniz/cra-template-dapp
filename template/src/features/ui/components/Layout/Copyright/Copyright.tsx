import { Box, Link, Button } from '@chakra-ui/react';
import { FaExternalLinkAlt } from '@react-icons/all-files/fa/FaExternalLinkAlt';
import React from 'react';

// You can remove or change this section
// however, I'd like to reach people as much as possible
// and I'd be appreceated if you don't remove it
// HÃ¼seyin Deniz KIVRAK - https://github.com/huseyindeniz

export const Copyright: React.FC = React.memo(() => {
  const copyrightLabel: string = 'dApp CRA Template';
  const copyrightUrl: string =
    'https://github.com/huseyindeniz/cra-template-dapp';

  return (
    <Box color="gray">
      <Button
        as={Link}
        href={copyrightUrl}
        rel="noopener noreferrer"
        isExternal
        variant="ghost"
        size="xs"
        rightIcon={<FaExternalLinkAlt />}
      >
        {copyrightLabel}
      </Button>
    </Box>
  );
});
