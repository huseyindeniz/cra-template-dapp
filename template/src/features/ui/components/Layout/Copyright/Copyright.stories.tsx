// ConnectButton.stories.ts|tsx
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Copyright } from './Copyright';
export default {
  title: 'ui/Components/Layout/Copyright',
  component: Copyright,
} as ComponentMeta<typeof Copyright>;

export const Default: ComponentStory<typeof Copyright> = args => (
  <Copyright {...args} />
);
