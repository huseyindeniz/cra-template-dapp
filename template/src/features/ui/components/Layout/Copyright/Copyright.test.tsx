import { render } from '@testing-library/react';

import { Copyright } from './Copyright';

describe('Feature: UI', () => {
  describe('Component: Layout/Copyright', () => {
    it('should be visible and show correct label and URL', () => {
      // Arrange
      const { asFragment, getByText } = render(<Copyright />);
      // Assert
      expect(getByText('dApp CRA Template v1.0.1')).toBeVisible();
      expect(asFragment).toMatchSnapshot();
    });
  });
});
