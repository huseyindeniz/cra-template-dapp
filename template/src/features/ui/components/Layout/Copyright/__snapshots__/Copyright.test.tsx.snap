// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: UI Component: Layout/Copyright should be visible and show correct label and URL 1`] = `[Function]`;
